shell/browser/native_window_mac.mm

```diff
@@ -235,6 +234,11 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
   [self setFrame:[[self superview] bounds]];
 }
 
+// Refs https://github.com/electron/electron/pull/30507
+// TODO: Enable this flag to match upstream
```

Comment:
What do we need to do to enable this? is there a tracking issue?

Reply:
https://github.com/electron/electron/pull/30507 incorrectly stated that https://chromium-review.googlesource.com/c/chromium/src/+/3069272 disabled this flag, but that CL only disabled DesktopCaptureMacV2, but kept WindowCaptureMacV2 enabled.  I added a tracking issue here:
https://github.com/electron/electron/issues/30682

Reply:
Sorry should have expanded in the commit. The root issue here is that the output of `NativeWindowMac::GetDesktopMediaID` was originally adjusted for https://chromium-review.googlesource.com/c/chromium/src/+/2709931 which tries to use a new backend for window capture based on `DesktopMediaId::window_id` https://source.chromium.org/chromium/chromium/src/+/main:content/browser/renderer_host/media/in_process_video_capture_device_launcher.cc;l=209-218. It looks like the duplicating of `DesktopMediaId::id` into `DesktopMediaId::window_id` was done to match the code style of aura window captures.

Since our desktop capturer api is based on chrome's `NativeDesktopMediaList` we had to follow suit so that `DesktopMediaId` from `NativeWindowMac::GetDesktopMediaID` and `DesktopCapturer::GetSources` matched, ref spec https://github.com/electron/electron/blob/be43996d35746551d60c01f3ac51502a299d1352/spec-main/api-desktop-capturer-spec.ts#L118-L146

But doing so caused a regression for use of `navigator.mediaDevices.getUserMedia` because the expected `DesktopMediaId` is different for this capture as mentioned in https://github.com/electron/electron/pull/30507. From a high level looks like we are mixing output of `NativeWindowMac::GetDesktopMediaID` for two different types of capture which should be addressed.